src/Storefront/Controller/GenericExpressController.php

```diff
@@ -46,6 +47,51 @@ public function __construct(
     ) {
     }
 
+    #[Route(
+        path: '/payone/express-checkout/create-session/{paymentMethodId}',
+        name: 'frontend.account.payone.express-checkout.generic.create-session',
+        options: ['seo' => false],
+        defaults: ['XmlHttpRequest' => true],
+        methods: ['GET']
+    )]
+    public function createSessionAction(SalesChannelContext $context, string $paymentMethodId): Response
```

Comment:
is this also used by Amazon-Pay and PayPal (legacy)?
if note, does it make sense to separate this into a custom-controller, or add additional checks for the payment-method?

Reply:
No, so far only used by PayPal v2. Since the same thing happens as in the “redirectAction”, I think it should stay in this controller. The difference is simply the response, so instead of redirect, the orderId is returned as JSON, which was needed for PayPal v2, but could just as well be used by others.

Reply:
Ah, but there is another difference, which is why I had not yet done this with the private function. In the `redirectAction` there is a try-catch around the `$this->salesChannelContextSwitcher->update` call. In the event of an exception, it is then forwarded to the checkout confirm page. That wouldn't have made sense in my `createSessionAction`, which is why I don't have the try-catch in there and if there is an exception, a status code >= 400 is returned and this is taken into account accordingly in the JavaScript.

Reply:
Understand. make sense. 

i checked the code of the redirect-action. 

The catch seems like not to be correct, because a formViolation should be always a `ConstraintViolationException`. 
I checked the code behind the switch-route - it can throw different exceptions of type `CartException`
We catch a throwable. 

I think it make sense to catch only these two types of exceptions.
And these exceptions can be catched in the redirect-action only.

i added a pr for this: https://github.com/PAYONE-GmbH/shopware-6/pull/341
(it goes into your branch)